src/db/delete.rs

```diff
@@ -16,20 +16,22 @@ enum CrateDeletionError {
 pub fn delete_crate(conn: &Connection, name: &str) -> Result<(), Error> {
     let crate_id = get_id(conn, name)?;
     delete_crate_from_database(conn, name, crate_id)?;
-    if let Some(s3) = s3_client() {
+
+    if let Some(backend) = s3_client() {
```

Comment:
nit: you changed the types back, but not the variable names. All this code is going to be refactored soon though, so no need to fix it unless you want to.
```suggestion
    if let Some(client) = s3_client() {
```